query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/insert/OngoingValues.java

```diff
@@ -38,4 +40,33 @@
   default RegularInsert value(@NonNull String columnName, @NonNull Term value) {
     return value(CqlIdentifier.fromCql(columnName), value);
   }
+
+  /**
+   * Sets values for multiple columns in one call, as in {@code INSERT INTO ... (key1, key2) VALUES
+   * (value1, value2)}.
+   *
+   * <p>If the map contains columns that had already been added to this statement, the previous
+   * entries are discarded and the new entries will be added at the end of the list.
+   *
+   * <p>Implementation note: this is a default method only for backward compatibility. The default
+   * implementation calls {@link #value(CqlIdentifier, Term)} in a loop; it should be overridden if
+   * a more efficient alternative exists.
+   */
+  @NonNull
+  default RegularInsert valuesByIds(@NonNull Map<CqlIdentifier, Term> newAssignments) {
```

Comment:
Seems like a better alternative when compared to `throw new UnsupportedOperationException` :)